server/src/daemon/service.rs

```diff
@@ -24,11 +24,33 @@ use super::{
     collection::Collection, error::ServiceError, prompt::Prompt, session::Session, Result,
 };
 
+#[derive(Debug)]
+pub struct ServiceManager {
+    sessions: HashMap<OwnedObjectPath, Session>,
+}
+
+impl ServiceManager {
+    pub fn new() -> Self {
+        Self {
+            sessions: HashMap::new(),
+        }
+    }
+
+    pub fn sessions(&self) -> &HashMap<OwnedObjectPath, Session> {
+        &self.sessions
+    }
+
+    pub fn set_sessions(&mut self, path: OwnedObjectPath, session: Session) {
```

Comment:
I would call it insert_session and the method can't be mutable otherwise you would need mutable access everywhere which would block the async tasks. Instead use a mutex/rwlock

Reply:
understood, will do.

Reply:
done.